### PR TITLE
Feature keep original extension

### LINKED ISSUE #149: Can it keep jpeg extension?
After upload jpeg and trigger Lambda, it would transfer jpeg to jpg, how could it do to keep jpeg extension?

Thank you

This is my config
```
{
  "bucket": 
 "myBucket",
  "reduce": {
      "directory": "./reduce",
      "prefix": "/",
      "quality": 80,
      "acl": "public-read"
  }
}
```

### PR DESCRIPTION
**Pull request by ysugimoto**
Fixes #149

When we use `ImageType`, which detects MimeType, the result of extension might be formatted like `JPG -> jpg`. To avoid it, I implemented that keeps original extension by file comes from S3. 

This feature works when `keepExtension` is `true` and default is `false` to keep backward compatibility.

### CODE REVIEW

File: bin/configtest [hunk 1]
```diff
@@ -43,16 +43,19 @@ var reset   = '\u001b[0m';
...
 
     var acl = config.acl;
-    stdout.write(magenta + "Global S3 ACL:          " + reset + (acl || "Not set") + "\r\n");
+    stdout.write(magenta + "Global S3 ACL:           " + reset + (acl || "Not set") + "\r\n");
 
     var cacheControl = config.cacheControl;
-    stdout.write(magenta + "Global S3 CacheControl: " + reset + (cacheControl || "Not set") + "\r\n");
+    stdout.write(magenta + "Global S3 CacheControl:  " + reset + (cacheControl || "Not set") + "\r\n");
+
+    var keepOriginalExtension = config.keepOriginalExtension;
```
Comment (1000ch) [inline bin/configtest]: I think `keepExtension` is slightly simpler than `keepOriginalExtension`. Any thoughts?

Comment (ysugimoto) [reply bin/configtest]: Right... I'll change name as you pointed.

**Comment by ysugimoto** [comment]
@kdybicz Hmm, sometimes some reducing tests are failed, but success on re-run tests.
I guess it is concurrency issue, Do you have any ideas?

File: test/image-data.js [hunk 2]
```diff
@@ -53,6 +53,14 @@ test("Build output path with prefix and suffix", t => {
     t.is(image.combineWithDirectory({directory: "d/e", prefix: "prefix-", suffix: "_suffix"}), "d/e/prefix-key_suffix.png");
 });
 
+test("Build output path with keep orignal extension", t => {
+    t.is(image.combineWithDirectory({directory: "d/e", keepExtension: true}), "d/e/key.png");
+});
+
+test("Build output path with suffix", t => {
```
Comment (kdybicz) [inline test/image-data.js]: looks like "Build output path with suffix" is duplicated, did you mean to add a suffix test with an keep extension parameter?

Comment (ysugimoto) [reply test/image-data.js]: Ah! sorry this duplication is my miss-copy

**Comment by ysugimoto** [comment]
Thanks! I merge this 😄

**Pull request merged**